src/Frontend.elm

```diff
@@ -176,6 +176,36 @@ update msg model =
             , Task.perform SetToday Date.today
             )
 
+        ShowEditTransactionDialog transactionId ->
+            case List.find (\t -> t.transactionId == transactionId) model.groupTransactions of
+                Just transaction ->
+                    let
+                        date = Date.fromCalendarDate transactionId.year (Date.numberToMonth transactionId.month) transactionId.day
+                        dateText = Date.format "yyyy-MM-dd" date
+                        total = transaction.total |> (\(Amount amount) -> amount) |> viewAmount
+                    in
+                    ( { model
+                        | showDialog =
+                            Just
+                                (EditTransactionDialog
+                                    { transactionId = transactionId
+                                    , description = transaction.description
+                                    , date = Just date
+                                    , dateText = dateText
+                                    , datePickerModel = DatePicker.init
+                                    , total = total
+                                    , credits = []
+                                    , debits = []
```

Comment:
These should also be pre-filled. This might require fetching more data from the backend.